crates/log-server/src/service.rs

```diff
@@ -52,52 +53,79 @@ impl LogServerService {
         metadata_store_client: MetadataStoreClient,
         record_cache: RecordCache,
         router_builder: &mut MessageRouterBuilder,
+        server_builder: &mut NetworkServerBuilder,
     ) -> Result<Self, LogServerBuildError> {
         describe_metrics();
         health_status.update(LogServerStatus::StartingUp);
 
-        let request_processor = RequestPump::new(updateable_config.clone(), router_builder);
+        // What do we need to create the log-server?
+        //
+        // 1. A log-store
+        let log_store_builder = RocksDbLogStoreBuilder::create(
+            updateable_config.clone().map(|c| &c.log_server).boxed(),
+            updateable_config
+                .clone()
+                .map(|c| &c.log_server.rocksdb)
+                .boxed(),
+            record_cache.clone(),
+        )
+        .await?;
+
+        // 2. Fire up the log store.
+        let log_store = log_store_builder.start(health_status.clone()).await?;
+
+        // Might fetch all known loglets from disk
+        let state_map = LogletStateMap::load_all(&log_store)
+            .await
+            .map_err(LogServerBuildError::LoadingLogletStateMap)?;
+
+        // 3. Register the log-server grpc service
+        server_builder.register_grpc_service(
+            TonicServiceFilter::new(
+                LogServerSvcServer::new(LogServerSvcHandler::new(
+                    log_store.clone(),
+                    state_map.clone(),
+                    record_cache,
+                ))
+                .accept_compressed(CompressionEncoding::Gzip)
+                .send_compressed(CompressionEncoding::Gzip),
+                {
+                    let health_status = health_status.clone();
+                    move |req| {
+                        if *health_status.get() == LogServerStatus::Ready {
```

Comment:
The good news is that we won't hit this with every message through the connection stream, so the lock contention on the watch should not impact the network layer once the connection/stream is created.